content/en/blog/_posts/2024-08-14-Kubernetes-v1-31-Release.md

```diff
@@ -0,0 +1,293 @@
+
+This feature allows webhook authorizers and future (but not currently designed) in-tree authorizers to allow list and watch requests based on field and label selectors.  It is now possible for an authorizer to express: this user cannot list all pods, but can list all pods where `.spec.nodeName=foo`.  Combined with CRD field selectors (moving to beta in 1.31), it is possible to write more secure per-node extensions.
+    
+This work was done as part of [KEP #4601](https://github.com/kubernetes/enhancements/issues/4601) by [SIG Auth](https://github.com/kubernetes/community/tree/master/sig-auth).
+    
+
+### Only allow anonymous auth for configured endpoints going Alpha in 1.31
+
+By enabling the feature gate `AnonymousAuthConfigurableEndpoints` users can now use the authentication configuration file to configure the endpoints that can be accessed by anonymous requests. This allows users to protect themselves against RBAC misconfigurations that can give anonymous users broad access to the cluster.
+    
+
+This work was done as a part of [KEP #4633](https://github.com/kubernetes/enhancements/issues/4633) and lead by [SIG Auth](https://github.com/kubernetes/community/tree/master/sig-auth).
+    
+## Graduations, deprecations, and removals in 1.31
+    
+
+### Graduations to Stable
+
+This lists all the features that graduated to stable (also known as _general availability_). For a full list of updates including new features and graduations from alpha to beta, see the release notes.
+
+This release includes a total of 11 enhancements promoted to Stable:
+
+* [PersistentVolume last phase transition time](	https://github.com/kubernetes/enhancements/issues/3762)
+* [Metric cardinality enforcement](	https://github.com/kubernetes/enhancements/issues/2305)
+* [Kube-proxy improved ingress connectivity reliability](	https://github.com/kubernetes/enhancements/issues/3836)
+* [Add CDI devices to device plugin API](	https://github.com/kubernetes/enhancements/issues/4009)
+* [Move cgroup v1 support into maintenance mode](	https://github.com/kubernetes/enhancements/issues/4569)
+* [AppArmor support](	https://github.com/kubernetes/enhancements/issues/24)
+* [PodHealthyPolicy for PodDisruptionBudget](	https://github.com/kubernetes/enhancements/issues/3017)
+* [Retriable and non-retriable Pod failures for Jobs](	https://github.com/kubernetes/enhancements/issues/3329)
+* [Elastic Indexed Jobs](	https://github.com/kubernetes/enhancements/issues/3715)
+* [Allow StatefulSet to control start replica ordinal numbering](	https://github.com/kubernetes/enhancements/issues/3335)
+* [Random Pod selection on ReplicaSet downscaling](	https://github.com/kubernetes/enhancements/issues/2185)
+    
+    
+### Deprecations and Removals 
+
+Also see our [Kubernetes Removals and Major Changes In v1.31](https://kubernetes.io/blog/2024/07/19/kubernetes-1-31-upcoming-changes/) blog.
+    
+#### Cgroup v1 enters the maintenance mode
```

Comment:
There is a blog going out https://github.com/kubernetes/website/pull/47069, currently set for 8/15. Should we move the date of that blog to go out the same day as the release? I'd like to make sure we don't create confusion about this topic so i think it would be good if we can link to this blog to give readers more context

Reply:
the content as written, currently links to the blog. So we'd have to update that if we dont publish the blogs the same day.

Reply:
@harche can you review this content around the cgroup maintenance mode for the release blog?

Reply:
@rashansmith  what do you think about the publishing date?

xref: https://github.com/kubernetes/enhancements/issues/4569#issuecomment-2182864829

Reply:
@harche, yes, lets move up the publication date to August 13th your [blog](https://github.com/kubernetes/website/pull/47069). It will go out the same day as the release so we can link readers to your blog for further reading.

Reply:
please also see https://github.com/kubernetes/website/pull/47069#issuecomment-2277953863. I've updated the content here to include more information in the release blog and no longer link to the https://github.com/kubernetes/website/pull/47069 post. This way the https://github.com/kubernetes/website/pull/47069 can go out on august 14th and we dont have to worry about timestamps. This release blog will be the only blog we publish August 13th and feature blogs will start publishing august 14th.

Reply:
We can _if we want_ change #47069, post release. The change I'm imagining would revise this article with a link to the new post.